### ISSUE TITLE
Mol to IDs is run 4 times, takes 2x more compute

### ISSUE THREAD

**Issue by DomInvivo** [datamol-io/graphium#122]
When running the code, you'll notice that the progress bar `mols to ids:` will appear 4 times:

1. Full set
2. Training set
3. Val set
4. Test set

This means that the mol-to-ids conversion is done twice per molecule. Although it is quite fast, it would be better to save their computation and re-use it.
